misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/TesterinaUtils.java

```diff
@@ -123,92 +110,48 @@ private static void execute(TestSuite suite, ClassLoader classLoader, TestArgume
-        TesterinaFunction configInit = new TesterinaFunction(configClazz, "$configureInit", initScheduler);
-        TesterinaFunction testExecute = new TesterinaFunction(testExecuteClazz, "__execute__", initScheduler);
-        // As the init function we need to use $moduleInit to initialize all the dependent modules
-        // properly.
-
-        Object response = configInit.directInvoke(new Class[]{String[].class, Path[].class, String.class},
-                new Object[]{new String[]{}, getConfigPaths(suite), null});
+    private static void startSuite(Class<?> initClazz, String[] args) {
+        // Call test module main
+        String funcName = cleanupFunctionName("main");
+        Object response = runTestModuleMain(initClazz, funcName, args, String[].class);
         if (response instanceof Throwable) {
-            throw new BallerinaTestException("configurable initialization for test suite failed due to " +
+            throw new BallerinaTestException("dependant module execution for test suite failed due to " +
                     formatErrorMessage((Throwable) response), (Throwable) response);
         }
+    }
 
-        init.setName("$moduleInit");
-        response = init.invoke();
-        if (response instanceof Throwable) {
-            throw new BallerinaTestException("dependant module initialization for test suite failed due to " +
-                    formatErrorMessage((Throwable) response), (Throwable) response);
-        }
-        // As the start function we need to use $moduleStart to start all the dependent modules
-        // properly.
-        start.setName("$moduleStart");
-        response = start.invoke();
-        if (response instanceof Throwable) {
-            throw new BallerinaTestException("dependant module start for test suite failed due to error : " +
-                    formatErrorMessage((Throwable) response), (Throwable) response);
-        }
+    private static Object runTestModuleMain(Class<?> initClazz, String name, String[] args,
+                                            Class<?>... parameterTypes) {
 
-        response = testExecute.invoke(args.getArgTypes(), args.getArgValues());
-        if (response instanceof Throwable) {
-            throw new BallerinaTestException();
+        try {
+            final Method method = initClazz.getDeclaredMethod(name, parameterTypes);
+            return method.invoke(null, (Object) args);
+        } catch (InvocationTargetException e) {
+            Throwable targetException = e.getTargetException();
+            if (targetException instanceof BError) {
+                return targetException;
+            }
+            return targetException;
+        } catch (NoSuchMethodException | IllegalAccessException | IllegalArgumentException e) {
+            return new BallerinaTestException("Failed to invoke the function '" + name + " due to " +
+                    e.getMessage(), e);
```

Comment:
Shall we check e.getmeesage is null
if null we can print e.getPrintablestacktrace

Reply:
We can use the `TesterinaUtils.formatErrorMessage(e)` method to append the stack trace of `e` along with the final `BallerinaTestException`. Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/39599/commits/333d13d3939bb35c15761fc40873506af6a73c59